### PR TITLE
Indices UI

### PR DESCRIPTION
**Pull request by agriffard**
Related to #4747

### CODE REVIEW

**Comment by agriffard** [comment]
image

File: src/OrchardCore.Modules/OrchardCore.Lucene/Controllers/AdminController.cs [hunk 1]
```diff
@@ -57,13 +69,31 @@ public class AdminController : Controller
...
         {
             var viewModel = new AdminIndexViewModel();
+            var siteSettings = await _siteService.GetSiteSettingsAsync();
+            var pager = new Pager(pagerParameters, siteSettings.PageSize);
 
             viewModel.Indexes = _luceneIndexManager.List().Select(s => new IndexViewModel { Name = s }).ToArray();
 
-            return View(viewModel);
+            var count = viewModel.Indexes.Count();
+
```
Comment (hishamco) [inline src/OrchardCore.Modules/OrchardCore.Lucene/Controllers/AdminController.cs]: Remove extra line

File: src/OrchardCore.Modules/OrchardCore.Lucene/Controllers/AdminController.cs [hunk 2]
```diff
@@ -57,13 +69,31 @@ public class AdminController : Controller
...
+            var results = viewModel.Indexes
+                .Skip(pager.GetStartIndex())
+                .Take(pager.PageSize).ToList();
+
```
Comment (hishamco) [inline src/OrchardCore.Modules/OrchardCore.Lucene/Controllers/AdminController.cs]: Remove extra line

File: src/OrchardCore.Modules/OrchardCore.Lucene/Controllers/AdminController.cs [hunk 3]
```diff
@@ -28,6 +32,9 @@ public class AdminController : Controller
         private readonly LuceneAnalyzerManager _luceneAnalyzerManager;
         private readonly ILuceneQueryService _queryService;
         private readonly ILiquidTemplateManager _liquidTemplateManager;
+        private readonly ISiteService _siteService;
+
```
Comment (hishamco) [inline src/OrchardCore.Modules/OrchardCore.Lucene/Controllers/AdminController.cs]: Remove extra line

File: src/OrchardCore.Modules/OrchardCore.Lucene/Controllers/AdminController.cs [hunk 4]
```diff
@@ -48,6 +57,9 @@ public class AdminController : Controller
             _queryService = queryService;
             _liquidTemplateManager = liquidTemplateManager;
             _notifier = notifier;
+            _siteService = siteService;
+
```
Comment (hishamco) [inline src/OrchardCore.Modules/OrchardCore.Lucene/Controllers/AdminController.cs]: Remove extra line

File: src/OrchardCore.Modules/OrchardCore.Lucene/Controllers/AdminController.cs [hunk 5]
```diff
@@ -57,13 +69,31 @@ public class AdminController : Controller
...
+            var count = viewModel.Indexes.Count();
+
+            var results = viewModel.Indexes
+                .Skip(pager.GetStartIndex())
+                .Take(pager.PageSize).ToList();
+
+            // Maintain previous route data when generating page links
+            var routeData = new RouteData();
+            var pagerShape = (await New.Pager(pager)).TotalItemCount(count).RouteData(routeData);
+
```
Comment (hishamco) [inline src/OrchardCore.Modules/OrchardCore.Lucene/Controllers/AdminController.cs]: Remove extra line

File: src/OrchardCore.Modules/OrchardCore.Lucene/OrchardCore.Lucene.csproj [hunk 6]
```diff
@@ -11,6 +11,7 @@
 
   <ItemGroup>
     <ProjectReference Include="..\..\OrchardCore\OrchardCore.Contents.Core\OrchardCore.Contents.Core.csproj" />
+    <ProjectReference Include="..\..\OrchardCore\OrchardCore.Localization.Abstractions\OrchardCore.Localization.Abstractions.csproj" />
```
Comment (hishamco) [inline src/OrchardCore.Modules/OrchardCore.Lucene/OrchardCore.Lucene.csproj]: Why?!!

File: src/OrchardCore.Modules/OrchardCore.Lucene/Views/Admin/Index.cshtml [hunk 7]
```diff
@@ -1,21 +1,31 @@
...
+            </div>
+        </div>
+    </div>
+    <ul class="list-group">
+        @if (Model.Indexes.Any())
+        {
+            <li class="list-group-item bg-light">
+                <div class="row">
+                    <div class="form-group col-xl-4 mb-xl-n1">
+                        <label id="items" for="select-all">@T.Plural(Model.Indexes.Count(), "1 item", "{0} items")<span class="text-muted" title="@T["Items {0} to {1}", startIndex, endIndex]">@T.Plural((int)Model.Pager.TotalItemCount, " / {0} item in total", " / {0} items in total")</span></label>
```
Comment (Skrypt) [inline src/OrchardCore.Modules/OrchardCore.Lucene/Views/Admin/Index.cshtml]: @hishamco probably because of this.

Comment (hishamco) [reply src/OrchardCore.Modules/OrchardCore.Lucene/Views/Admin/Index.cshtml]: You 're right .. thanks

**Pull request merged**